crates/bevy_math/src/curve/mod.rs

```diff
@@ -0,0 +1,610 @@
+//! The [`Curve`] trait, used to describe curves in a number of different domains. This module also
+//! contains the [`Interval`] type, along with a selection of core data structures used to back
+//! curves that are interpolated from samples.
+
+pub mod interval;
+
+pub use interval::{everywhere, interval, Interval};
+
+use interval::{InfiniteIntervalError, InvalidIntervalError};
+use std::{marker::PhantomData, ops::Deref};
+use thiserror::Error;
+
+#[cfg(feature = "bevy_reflect")]
+use bevy_reflect::Reflect;
+
+/// A trait for a type that can represent values of type `T` parametrized over a fixed interval.
+/// Typical examples of this are actual geometric curves where `T: VectorSpace`, but other kinds
+/// of interpolable data can be represented instead (or in addition).
+pub trait Curve<T> {
+    /// The interval over which this curve is parametrized.
+    fn domain(&self) -> Interval;
+
+    /// Sample a point on this curve at the parameter value `t`, extracting the associated value.
```

Comment:
The meaning of `t` (0 = start of domain, 1 = end of domain) should be explained in the trait-level docs.

Reply:
When I understood it correct, t = 0 doesn't mean the start of the domain, but rather just 0. The domain can, but doesn't have to be 0..=1

Reply:
otherwise, `Curve::domain` would be useless 😨

Reply:
The documentation for `domain` now uses `t`, mirroring its use in `sample` and `sample_unchecked`.